### PR TITLE
test(engine):add DM test case new_collation_off

### LINKED ISSUE #6851: executor panic after task resume
### What did you do?

[df.log](https://github.com/pingcap/tiflow/files/9399245/df.log)

### What did you expect to see?

_No response_

### What did you see instead?

```
3m3e-server-executor-2               | panic: duplicate metrics collector registration attempted
```

### Versions of the cluster

Dataflow Engine version (run `tiflow version`):

```console
(paste Dataflow Engine version here)
```

### PR DESCRIPTION
**Pull request by lance6716**
### What problem does this PR solve?

Issue Number: close #6851 ref #6776 

#### Tests 

 - Integration test
 - Manual test

### Release note 

```release-note
`None`.
```

### CODE REVIEW

**Comment by lance6716** [comment]
/run-engine-integration-test

PTAL @CharlesCheung96 @maxshuang

File: dm/loader/lightning.go [hunk 1]
```diff
@@ -216,16 +216,18 @@ func (l *LightningLoader) runLightning(ctx context.Context, cfg *lcfg.Config) er
 	var opts []lightning.Option
 	if l.cfg.MetricsFactory != nil {
 		// this branch means dataflow engine has set a Factory, the Factory itself
-		// will register and deregister metrics, so we must use NoopRegistry
-		// to avoid duplicated registration.
+		// will register and deregister metrics, but lightning will expect the
```
Comment (maxshuang) [inline dm/loader/lightning.go:219]: Do you think we need to find a better way for registering and unregistering the metric?

File: deployments/engine/docker-compose/dm_databases_tidb_new_collation_off.yaml [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+        ports:
+            - "3306:3306"
+        volumes:
+            - ./config/mysql.cnf:/etc/mysql/conf.d/mysql.cnf
+        environment:
+            MYSQL_ALLOW_EMPTY_PASSWORD: "yes"
+
+    dm_upstream_mysql2:
+        image: mysql:8.0
+        container_name: dm_upstream_mysql2
```
Comment (CharlesCheung96) [inline deployments/engine/docker-compose/dm_databases_tidb_new_collation_off.yaml:16]:
```suggestion
        container_name: dm_upstream_mysql2
        command: --default-authentication-plugin=mysql_native_password
```

**Review by GMHDBJD — APPROVED** [review]
LGTM

**Comment by maxshuang** [comment]
Rest LGTM.

File: deployments/engine/docker-compose/dm_databases_tidb_new_collation_off.yaml [hunk 3]
```diff
@@ -0,0 +1,61 @@
+version: '2.3'
```
Comment (CharlesCheung96) [inline deployments/engine/docker-compose/dm_databases_tidb_new_collation_off.yaml:1]: Better to use a general filename, such as: `dm_database_tidb_v6.1.yaml`

Comment (lance6716) [reply deployments/engine/docker-compose/dm_databases_tidb_new_collation_off.yaml:1]: new_collation_off is the key difference of this deployment. (I'll change v6.1 to nightly in next commit)

**Review by CharlesCheung96 — APPROVED** [review]

**Pull request merged**